src/transformers/models/speecht5/modeling_speecht5.py

```diff
@@ -2242,7 +2243,7 @@ def forward(
     """SpeechT5 Model with a speech encoder and a text decoder.""",
     SPEECHT5_START_DOCSTRING,
 )
-class SpeechT5ForSpeechToText(SpeechT5PreTrainedModel):
+class SpeechT5ForSpeechToText(SpeechT5PreTrainedModel, GenerationMixin):
```

Comment:
`SpeechT5ForSpeechToText` probably had some `generate` compatibility issues in the last versions, since its `prepare_inputs_for_generation` was not being updated

(removing the global mixin triggered test failures)